src/evaluate/harness/__init__.py

```diff
@@ -0,0 +1,71 @@
+import json
+import os
+
+from evaluate import evaluator
+from evaluate.loading import convert_hf_hub_path, relative_to_absolute_path
+from evaluate.utils.file_utils import cached_path, is_relative_path
+from evaluate.utils.logging import get_logger
+
+
+logger = get_logger(__name__)
+
+
+class Harness:
+    """
+    This class instantiates an evaluation harness made up of multiple tasks, where each task consists of a dataset and
+    an associated metric, and runs evaluation on a model or pipeline. Harnesses can be instantiated from a JSON file
+    named harness_config.json found either locally or uploaded as a dataset on the Hugging Face Hub.
+
+    Usage:
+    ```python
+    >>> from evaluate.harness import Harness
+
+    >>> harness = Harness('mathemakitten/glue-harness')
+    >>> results = harness.run(model_or_pipeline='gpt2')
+    ```
+    """
+
+    def __init__(self, path):
+        """Instantiates a Harness object from a JSON file which will be passed to the Evaluator to run evaluation for a
+        model on this collection of tasks."""
+
+        filename = list(filter(lambda x: x, path.replace(os.sep, "/").split("/")))[-1]
```

Comment:
does this work if the user only provides the folder to the harness json? e.g. if I pass `"~/lvwerra/glue-harness"` which is the folder that contains the json file.

Reply:
Now it does, good catch :)